actions/generate-schedule/main.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/spinnaker/rotation-scheduler/actions/generate-schedule/scheduler"
+	"github.com/spinnaker/rotation-scheduler/actions/generate-schedule/users"
+	"github.com/spinnaker/rotation-scheduler/proto/schedule"
+)
+
+var (
+	// Required flags
+	stop              = flag.String("stop", "", "Stop generating the schedule after this date. Date is exclusive (not included in the schedule), and must be in the format 2006-01-02.")
+	shiftDurationDays = flag.Int("shiftDurationDays", 7, "The duration of each shift in days")
+
+	// One of these flags is required
+	start                = flag.String("start", "", "Start generating from this date. If this flag and previousSchedule are not specified, starts from tomorrow's date.")
+	previousSchedulePath = flag.String("previousSchedule", "", "Path to a previous schedule's YAML file.")
+
+	// TODO(ttomsu): Flag is required now, make optional later (once GH teams integration works).
+	userList = flag.String("users", "", "Comma-separated list of users in the rotation. Usernames will be alphabetically sorted prior to assigning shifts.")
+
+	outFile = flag.String("out", "", "File to write out new schedule. Will overwrite the file if it exists.")
+)
+
+const (
+	startStopFormat = "2006-01-02"
+	usersSeparator  = ","
+)
+
+func main() {
+	flag.Parse()
+
+	if err := validateFlags(); err != nil {
+		log.Fatalf("Error validating flags: %v", err)
+	}
+
+	userSlice := strings.Split(*userList, usersSeparator)
+	userSource := users.NewStaticSource(userSlice...)
+
+	sched, err := scheduler.NewScheduler(userSource, *shiftDurationDays)
+	if err != nil {
+		log.Fatalf("Error creating scheduler: %v", err)
+	}
+
+	stopTime, _ := time.Parse(startStopFormat, *stop)
+
+	var newSchedule *schedule.Schedule
+	if *previousSchedulePath != "" {
+		previousSchedule, err := parsePreviousSchedule(*previousSchedulePath)
+		if err != nil {
+			log.Fatalf("Error parsing previous schedule: %v", err)
+		}
+
+		newSchedule, err = sched.ExtendSchedule(previousSchedule, stopTime)
+		if err != nil {
+			log.Fatalf("Error extending schedule: %v", err)
+		}
+	} else {
+		var startTime time.Time
+		if *start != "" {
+			startTime, _ = time.Parse(startStopFormat, *start)
+		} else {
+			startTime = time.Now().AddDate(0, 0, 1) // tomorrow.
```

Comment:
Should we truncate this to be only today's date?

Reply:
This is mostly for bootstrapping. I figured that anybody bootstrapping this would a.) be doing this manually, and b.) doing it sometime midday, so the first full schedule would happen the following day.

Since it's just bootstrapping I can chomp it down to just today if you feel strongly about it.

Reply:
Oh, I didn't mean to remove the `AddDate`, just to ignore the time part of the date.  I *think* that gets dropped somewhere downstream anyway (when we convert the date to a string using the format) but just found it a little odd that we had an exact date-only value in one block of the `if` and a date+time in the other.

Reply:
Yeah, at no point is the _time_ component used, but the whole `time` package revolves around date+time. `time.Now()` is just shorter than manually creating a `Time` object